# circularMT-console

This program is based on the desktop application [```circularMT```](https://github.com/msjimc/circularMT). Since ```circularMT``` has a interactive graphical user interface it is more flexible than circularMT-console. 

## Contents

- [Introduction](#Introduction)
- [Guide](#guide)
- [Download](#download)
- [Running on Linux systems](#running-on-linux-systems)

## Introduction

<img align="right" src="../Guide/images/introCircular.jpg">

```circularMT-console``` is designed to create images of the genomic organisation of mitochondrial genomes that can be used to display the arrangement of the genes they encode. Images can be exported as 300 dpi TIFF, bitmap, PNG or JPEG files for use in reports, publications or a thesis. The maps can be circular like the one to the right or linear like the one below.

The data can be imported from a wide range of file formats such as *.fasta, *.mitos, *.gb, *.bed, *.seq, *.gtf and *.gff files. While ```circularMT``` can process these files, *.fasta and *.bed files in particular most conform to a set format not present in all files.

The program is based on the desktop circularMT however, due to the lack of user interaction possible with command line applications compared to desktop programs, there is less opportunity to tidy up and image.

<center><img src="../Guide/images/introLineear.jpg"></center>

## Guide

The user guide is [here](Guide/README.md).  

Example data files are in the circularMT > [Example data](../Example%20data/) folder.

## Download

To download the source code etc, clone the whole circularMT repo and then open the circularMT-console project in Visual STudio by clicking on the circularMT-console.sln file in the the circularMT > circularMT-console folder.

The prebuilt program can be downloaded [here](Program/README.md).

## Running on Linux systems

The prebuilt programs can be run on Linux (and macOS) using the Wine package as described in the circularMT > [Linux with Wine](../Linux_with_Wine/README.md) readme file.






